### ISSUE TITLE
ImageMagick timeouts.

### ISSUE THREAD

**Issue by Danack** [gographics/imagick#71]
Hello,

I'm the maintainer of the PHP extension for Imagick. This is a slight 'drive by' giving of information, to give you guys a heads up about an issue that I encountered with using ImageMagick in a long-running process.

The original issue is [here](https://github.com/ImageMagick/ImageMagick/issues/113)

The feature request for the ImageMagick guys to make it be not a problem is here:
https://www.imagemagick.org/discourse-server/viewtopic.php?f=2&t=29083&p=129885#p129885

Short version - any process that is using the ImageMagick library needs to restart more frequently than the "max image processing time" limit set in ImageMagick's config file.

Also, it would have been better to chose a different name for the binding.....that doesn't clash with another binding for another language.....I guess it's too late for that now though.

cheers
Dan

**Comment by justinfx** [comment]
Hi Dan,

Much appreciated for the heads up. On the note about the name of the binding, I apologize that it conflicts with previously existing bindings. Too late in the game to change it (and also, I didn't pick the name; I am just a maintainer).

Let me see if I understand this issue properly. In my local install of im 6.9.1.1, I have an entry in the policy.xml like the following:

```
  <!-- <policy domain="resource" name="time" value="3600"/> -->
```

Are you saying that this value imposes a global 60 minute limit from the time the first image is processed, to where future image processing would then fail? I surely hope that is not true, because this doesn't sound like a feature request. It sounds like a huge flaw/bug. 
Or is it only activated when accessing the `MagickSetResourceLimit ` api call to set the time resource?
Interestingly, when I call our `GetResourceLimit(imagick.RESOURCE_TIME)` function, it reports a value of -1, regardless of that config value.

**Comment by Danack** [comment]
> Are you saying that this value imposes a global 60 minute limit from the time the first image is processed, 

Yes, or rather would if it wasn't commented out by the XML 

I noticed the issue when I set a time limit of 45 seconds (which is far more than I need for the images I'm processing), and the background image processor I use started failing constantly, 45 seconds after processing the first image.

> I surely hope that is not true, because this doesn't sound like a feature request. It sounds like a huge flaw/bug. 

Po-tay-to, po-tah-to.
To-may-to, glo-bal nu-ku-lar war-fare.

They are working on fixing it apparently, but yes - if your users are likely to be running apps that run continuously (which is different from PHP running in PHP-FPM where everything gets reloaded every X requests) then they would almost certainly encounter this issue.

**Comment by justinfx** [comment]
I personally used to run an image cache server which would resize original
images and then server them out of a cache for future requests. I never set
the time resource so I had never run into this.
I will keep this in mind for the future, should anyone report a related
problem. Thanks again for the heads up!

**Comment by Danack** [comment]
No problem.
